AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_entity_utils.py

```diff
@@ -50,6 +62,9 @@ def get_property_tree(self):
             7. prop_tree.get_container_item(path, key)
         :return: Property tree object of a component
         """
+        if self.property_tree is not None:
+            return self.property_tree
```

Comment:
We might want to add an optional parameter to this method to force get a new property tree because the values could get updated by other ebus calls.

Reply:
good call. I've added force_get: bool to address this situation.

Reply:
May want to further clarify that this method is caching the property tree in the first line of the docstring

Reply:
In fact I'm debating whether the user should be left to handle the property_tree returned by this method, instead of relying on this class to cache on their behalf. Without a major change to this code, the methods below could accept a property tree from the user instead of relying on the cache.

However, what is the argument against the methods below always fetching a fresh property tree? Is it expensive?

Reply:
The design intention was that the consumer of this utility could not be concerned with property tree editor at all. they could instantiate a new component then directly attempt to append to a property container or get the count of properties in a container. previous to this they would have to get a PTE then write all the code to handle it locally in their test. Duplicating common code in several tests.
The code could be written to always get a new PTE for every call to a container method, but that would involve several ebus calls for every action. the default force_get could be True as well which would essentially be always get a fresh PTE.

Reply:
I think the naming refactor would clear that up for anyone who needs maintain this code. Also, in the case of the caching now occurring, I suggest not even passing back self.property_tree(_editor) at the end of this method, as it suggests the caller *should* be concerned with it.

Reply:
existing users of the utility have implemented test code with this method so changing the signature of get_property_tree itself would be breaking for anyone who expects to manipulate a PTE returned from this method. Ideally this would be marked as an informally private method with initial underscore but people currently use it as public. The intention of caching to self.property_tree_editor was so that class methods that use PTE would not have to create a new PTE for each call.